clinica/iotools/bids_utils.py

```diff
@@ -523,168 +534,247 @@ def write_modality_agnostic_files(
     following the BIDS specification.
 
-    Args:
-        bids_dir: path to the bids directory
-        sessions_dict: output of the function create_sessions_dict
+    Parameters
+    ----------
+    bids_dir : str
+        Path to the BIDS directory.
+
+    sessions_dict : dict
+        Dictionary containing sessions metadata.
+
+        .. note::
+            This is the output of the function
+            `clinica.iotools.bids_utils.create_sessions_dict`.
+
+    See also
+    --------
+    create_sessions_dict
+    write_scans_tsv
     """
-    import os
-    from glob import glob
-    from os import path
+    from pathlib import Path
 
     import pandas as pd
 
-    bids_paths = glob(path.join(bids_dir, "sub-*"))
-
-    for sp in bids_paths:
-        bids_id = sp.split(os.sep)[-1]
+    bids_dir = Path(bids_dir)
 
-        if bids_id in sessions_dict:
-            session_df = pd.DataFrame.from_dict(sessions_dict[bids_id], orient="index")
+    for sp in bids_dir.glob("sub-*"):
+        if sp.name in sessions_dict:
+            session_df = pd.DataFrame.from_dict(sessions_dict[sp.name], orient="index")
             cols = session_df.columns.tolist()
             cols = cols[-1:] + cols[:-1]
             session_df = session_df[cols]
         else:
             print(f"No session data available for {sp}")
             session_df = pd.DataFrame(columns=["session_id"])
             session_df["session_id"] = pd.Series("M00")
-
         session_df = session_df.set_index("session_id").fillna("n/a")
-        session_df.to_csv(
-            path.join(sp, bids_id + "_sessions.tsv"),
-            sep="\t",
-            encoding="utf8",
+        session_df.to_csv(sp / f"{sp.name}_sessions.tsv", sep="\t", encoding="utf8")
+
+
+def _get_pet_tracer_from_filename(filename: str) -> str:
+    """Return the PET tracer from the provided filename.
+
+    Parameters
+    ----------
+    filename : str
+        The filename from which to extract the PET tracer.
+
+    Returns
+    -------
+    tracer : str
+        The PET tracer.
+
+    Raises
+    ------
+    ValueError
+        If no tracer found in the filename.
+    """
+    import re
+
+    tracer = None
+    for entity in ("trc", "acq"):
+        m = re.search(rf"({entity}-[a-zA-Z0-9]+)", filename)
+        if m:
+            tracer = m.group(1)[4:].upper()
+    if tracer is None:
+        raise ValueError(
+            f"Could not extract the PET tracer from the following file name {filename}."
         )
 
+    return tracer
+
 
-def write_scans_tsv(bids_dir, bids_ids, scans_dict):
+def write_scans_tsv(bids_dir: str, bids_ids: List[str], scans_dict: dict) -> None:
     """Write the scans dict into TSV files.
 
-    Args:
-        bids_dir:  path to the BIDS directory
-        bids_ids: list of bids ids
-        scans_dict:  the output of the function create_scans_dict
+    Parameters
+    ----------
+    bids_dir : str
+        Path to the BIDS directory.
+
+    bids_ids : List[str]
```

Comment:
Suggest to replace `bids_id` with `participant_id` wherever we can. The former makes no sense.

```suggestion
    participant_ids : List[str]
```